packages/connect/src/api/solana/api/solanaSignTransaction.ts

```diff
@@ -41,6 +71,159 @@ export default class SolanaSignTransaction extends AbstractMethod<'solanaSignTra
+                }
+                // Transfer instruction decoding
+                if (
+                    instruction.programAddress === SYSTEM_PROGRAM_ADDRESS &&
+                    instruction.data &&
+                    instruction.accounts
+                ) {
+                    const instructionSafe = {
+                        ...instruction,
+                        data: instruction.data as Uint8Array,
+                        accounts: instruction.accounts,
+                    };
+                    const type = identifySystemInstruction(instructionSafe);
+                    if (type === SystemInstruction.TransferSol) {
+                        const parsed = parseTransferSolInstruction(instructionSafe);
+                        outputs.push({
+                            address: parsed.accounts.destination.address,
+                            amount: parsed.data.amount.toString(),
+                            script_type: 'PAYTOADDRESS' as const,
+                        });
+                        sendAmount = sendAmount.plus(parsed.data.amount.toString());
+                    } else if (type === SystemInstruction.CreateAccount) {
+                        const parsed = parseCreateAccountInstruction(instructionSafe);
+                        outputs.push({
+                            address: parsed.accounts.newAccount.address,
+                            amount: parsed.data.lamports.toString(),
+                            script_type: 'PAYTOADDRESS' as const,
+                        });
+                        rent = rent.plus(parsed.data.lamports.toString());
+                    }
+                }
+                // Tokens
+                if (
+                    instruction.programAddress === TOKEN_PROGRAM_ADDRESS &&
+                    instruction.data &&
+                    instruction.accounts
+                ) {
+                    const instructionSafe = {
+                        ...instruction,
+                        data: instruction.data as Uint8Array,
+                        accounts: instruction.accounts,
+                    };
+                    const type = identifyTokenInstruction(instructionSafe);
+                    if (type === TokenInstruction.TransferChecked) {
+                        const parsed = parseTransferCheckedInstruction(instructionSafe);
+                        const destinationATA = parsed.accounts.destination.address;
+                        const tokenInfo = this.payload.additionalInfo?.tokenAccountsInfos?.find(
+                            t =>
+                                t.tokenAccount === destinationATA &&
+                                t.tokenMint === parsed.accounts.mint.address,
+                        );
+                        if (!sendAmount.isZero()) {
+                            throw new Error(
+                                'Multiple token transfers in a single transaction are not supported',
+                            );
```

Comment:
as a convention, we tend to use a typed error in connect. one of the list https://github.com/trezor/trezor-suite/blob/develop/packages/connect/src/constants/errors.ts#L33 or the "runtime" if its is an unexpected one.